clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3115,12 +3153,10 @@ static void handleWorkGroupSize(Sema &S, Decl *D, const ParsedAttr &AL) {
 
 // Handles work_group_size_hint.
 static void handleWorkGroupSizeHint(Sema &S, Decl *D, const ParsedAttr &AL) {
-  uint32_t WGSize[3];
-
-  if (!AL.checkExactlyNumArgs(S, 3))
-    return;
+  S.CheckDeprecatedSYCLAttributeSpelling(AL);
 
-  for (unsigned i = 0; i < 3; ++i) {
+  uint32_t WGSize[3];
+  for (unsigned i = 0; i < AL.getNumArgs(); ++i) {
     if (!checkUInt32Argument(S, AL, AL.getArgAsExpr(i), WGSize[i], i,
```

Comment:
I think you might need similar change like this so that the attribute will accept one, two, and three arg form.

  for (unsigned i = 0; i < 3; ++i) {
    if (i < AL.getNumArgs() &&
       !checkUInt32Argument(S, AL, AL.getArgAsExpr(i), WGSize[i], i,
                              /*StrictlyUnsigned=*/true))
       return;

// FIXME: the attribute is like reqd_work_group_size in that it has a one, two,
// and three arg form that needs to be supported.
[[sycl::work_group_size_hint(1)]] void f4(); // expected-error {{'work_group_size_hint' attribute requires exactly 3 arguments}}
[[sycl::work_group_size_hint(1, 1)]] void f5(); // expected-error {{'work_group_size_hint' attribute requires exactly 3 arguments}}

Reply:
I think we should handle the argument changes for `reqd_work_group_size` and `work_group_size_hint` at the same time, so for this patch, I've left the argument handling bits alone. Do you prefer I do those changes as part of this patch?

Reply:
> I think we should handle the argument changes for `reqd_work_group_size` and `work_group_size_hint` at the same time, so for this patch, I've left the argument handling bits alone. Do you prefer I do those changes as part of this patch?

>>I think we should handle the argument changes for `reqd_work_group_size` and `work_group_size_hint` at the same time, so for this patch, I've left the argument handling bits alone. 

I agree. We should do this in follow-up patch (do not need to make any changes as part of this patch here). Thanks.